src/iterable.cr

```diff
@@ -33,4 +33,40 @@ module Iterable
   def each_with_object(obj)
     each.with_object(obj)
   end
+
+  # :nodoc:
+  # Overrides Comparable.==
+  # This version works with more objects because it doesn't rely on <=>.
+  def ==(other: self)
+    # TODO: if T is a noncontainer value && Enumerable is allocated as an array use memcmp()
+    equals?(other) { |x, y| x == y }
+  end
```

Comment:
Perhaps we should fix `Comparable#==` or more generally `Object#<=>` instead. Ruby defines `Object#<=>` as returning `nil`:
> Your implementation of <=> should return one of the following values: -1, 0,
1 or nil. -1 means self is smaller than other. 0 means self is equal to other.
1 means self is bigger than other. Nil means the two values could not be
compared.